### DISCUSSION TITLE
Using arrays as queryKey?

### DISCUSSION THREAD

**Discussion by dhanushkac** [TanStack/query#4006] [Q&A]
Hi,

Can we use `['id', [1,2,3]]` as a proper queryKey? Or should we use `['id', [1,2,3]].flat()` to make it equivalent with `['id', 1,2,3]`?

**Comment by TkDodo** [comment]
it doesn't need to be flat - `['id', [1,2,3]]` is fine, but it will not be the same as `['id', [1,3,2]]`, but this should be obvious because arrays have a fixed order.
